src/antsibull/tagging.py

```diff
@@ -0,0 +1,135 @@
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or
+# https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: 2022 Maxwell G <gotmax@e.email>
+
+"""
+Validate that collections tag their releases in their respective git repositories
+"""
+import asyncio
+import os
+import re
+import typing as t
+
+import asyncio_pool  # type: ignore[import]
+
+from antsibull_core.logging import log
+
+from antsibull_core import app_context
+from antsibull_core.dependency_files import DepsFile
+from antsibull_core.yaml import store_yaml_file, load_yaml_file
+
+from antsibull.collection_meta import CollectionsMetadata
+
+mlog = log.fields(mod=__name__)
+
+
+def validate_tags_command() -> int:
+    app_ctx = app_context.app_ctx.get()
+    if app_ctx.extra['input']:
+        tag_data = load_yaml_file(app_ctx.extra['input'])
+    else:
+        tag_data = asyncio.run(get_collections_tags())
+        if app_ctx.extra['output']:
+            store_yaml_file(app_ctx.extra['output'], tag_data)
+    return validate_tags(tag_data)
+
+
+def validate_tags(tag_data: t.Dict[str, t.Dict[str, t.Optional[str]]]) -> int:
+    r = 0
```

Comment:
Shouldn't this function rather return a boolean instead of an integer? (Also it should have a docstring, similar to the other non-private functions in this file.)

Reply:
Sounds good as well.